### PR TITLE
[Python | * DateTimeV2|Number] Update base datetime and merged number extractor for closer .NET parity

### PR DESCRIPTION
**Pull request by samhickey25**

### CODE REVIEW

**Comment by samhickey25** [comment]
Updates to the resources were generate by running the script to generate them from the Patterns files

**Comment by samhickey25** [comment]
@tellarin Would it be possible for this to be reviewed?

**Comment by tellarin** [comment]
@samhickey25 This is next on the list. But it has some pending conflicts to resolve too (won't block review).

File: Python/libraries/recognizers-date-time/recognizers_date_time/date_time/german/date_parser_config.py [hunk 1]
```diff
@@ -193,9 +193,9 @@ def get_swift_day(self, source: str) -> int:
             swift = 0
         elif trimmed_text == 'morgen':
             swift = 1
-        elif trimmed_text == 'übermorgen':
+        elif trimmed_text == 'gestern':
```
Comment (tellarin) [inline Python/libraries/recognizers-date-time/recognizers_date_time/date_time/german/date_parser_config.py:196]: Ideally these should be changed to use YAML resources. Could you do it in a future PR?

File: Python/libraries/recognizers-date-time/recognizers_date_time/resources/english_date_time.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ class EnglishDateTime:
     PrefixDayRegex = f'\\b((?<EarlyPrefix>earl(y|ier))|(?<MidPrefix>mid(dle)?)|(?<LatePrefix>later?))(\\s+in)?(\\s+the\\s+day)?$'
     SeasonDescRegex = f'(?<seas>spring|summer|fall|autumn|winter)'
     SeasonRegex = f'\\b(?<season>({PrefixPeriodRegex}\\s+)?({RelativeRegex}\\s+)?{SeasonDescRegex}((\\s+of|\\s*,\\s*)?\\s+({YearRegex}|{RelativeRegex}\\s+year))?)\\b'
-    WhichWeekRegex = f'\\b(week)(\\s*)(?<number>5[0-3]|[1-4]\\d|0?[1-9])\\b'
+    WhichWeekRegex = f'\\b(week)(\\s*)(?<number>5[0-3]|[1-4]\\d|0?[1-9])(\\s+of\\s+({YearRegex}|{RelativeRegex}\\s+year))?\\b'
```
Comment (tellarin) [inline Python/libraries/recognizers-date-time/recognizers_date_time/resources/english_date_time.py:103]: I assue these are just updates from auto regenerated resources, correct? I don't see changes in YAML files and the build should be re-creating these and still passing.

**Review by tellarin — APPROVED** [review]

**Pull request merged**